### PR TITLE
AI/Cyborgs no longer need to directly click the door sprite to interact with it.

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About The Pull Request
AI/Cyborgs no longer have to directly press the sprite of the door to use their shortcut bolt/close/emergency acces
Thanks  to Bob Joga for helping with the code.
## Why It's Good For The Game
QoL for any AI/Cyborg player , there is no longer a need to shift the mouse a few pixels to close a door
## Changelog
:cl:
add:AI/Cyborgs no longer are required to click the door sprite to close/bolt/emergency acces them.
/:cl:

### CODE REVIEW

File: code/_onclick/cyborg.dm [hunk 1]
```diff
@@ -105,10 +105,26 @@
...
 /mob/living/silicon/robot/CtrlShiftClickOn(atom/A)
 	A.BorgCtrlShiftClick(src)
+	var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A // Skyrat edit
+	if(airlock)
+		airlock.AICtrlShiftClick(src)
+	return // End of skyrat edit
+
 /mob/living/silicon/robot/ShiftClickOn(atom/A)
 	A.BorgShiftClick(src)
+	A.AIShiftClick(src)
```
Comment (Azarak) [inline code/_onclick/cyborg.dm:115]:
```suggestion
```

File: code/_onclick/cyborg.dm [hunk 2]
```diff
@@ -105,10 +105,26 @@
 // for non-doors/apcs
 /mob/living/silicon/robot/CtrlShiftClickOn(atom/A)
 	A.BorgCtrlShiftClick(src)
+	var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A // Skyrat edit
+	if(airlock)
+		airlock.AICtrlShiftClick(src)
```
Comment (Azarak) [inline code/_onclick/cyborg.dm:110]:
```suggestion
		airlock.BorgCtrlShiftClick(src)
```

File: code/_onclick/cyborg.dm [hunk 3]
```diff
@@ -105,10 +105,26 @@
...
+	return // End of skyrat edit
+
 /mob/living/silicon/robot/ShiftClickOn(atom/A)
 	A.BorgShiftClick(src)
+	A.AIShiftClick(src)
+	var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A // Skyrat edit
+	if(airlock)
+		airlock.AIShiftClick(src)
```
Comment (Azarak) [inline code/_onclick/cyborg.dm:118]:
```suggestion
		airlock.BorgShiftClick(src)
```

File: code/_onclick/cyborg.dm [hunk 4]
```diff
@@ -105,10 +105,26 @@
...
+	return // End of skyrat edit
+
 /mob/living/silicon/robot/CtrlClickOn(atom/A)
 	A.BorgCtrlClick(src)
+	var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A // Skyrat edit
+	if(airlock)
+		airlock.AICtrlClick(src)
```
Comment (Azarak) [inline code/_onclick/cyborg.dm:125]:
```suggestion
		airlock.BorgCtrlClick(src)
```

**Review by Azarak — CHANGES_REQUESTED** [review]
Also, first check for the silicons to clicking on top of a turf, then check the airlock locate, and if they hasn't found an airlock to interact with, do the normal interaction with the atom.

File: code/_onclick/cyborg.dm (hunk 1 above)
Comment (MLGTASTICa) [reply code/_onclick/cyborg.dm]: Removed that , pasted it there by accident.

File: code/_onclick/cyborg.dm (hunk 2 above)
Comment (MLGTASTICa) [reply code/_onclick/cyborg.dm]: Done.

**Review by Gandalf2k15 — APPROVED** [review]

File: code/_onclick/ai.dm [hunk 5]
```diff
@@ -109,11 +109,29 @@
 */
 
 /mob/living/silicon/ai/CtrlShiftClickOn(var/atom/A)
-	A.AICtrlShiftClick(src)
+	if(isturf(A)) // Skyrat edit
+		var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A 
+		if(airlock)
+			airlock.AICtrlShiftClick(src)
+		A.AICtrlShiftClick(src)
+	A.AICtrlShiftClick(src) // End of skyrat edit
```
Comment (Azarak) [inline code/_onclick/ai.dm:117]:
```suggestion
	if(isturf(A)) // Skyrat edit
		var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A 
		if(airlock)
			airlock.AICtrlShiftClick(src)
	else
		A.AICtrlShiftClick(src) // End of skyrat edit
```

File: code/_onclick/ai.dm [hunk 6]
```diff
@@ -109,11 +109,29 @@
...
+
 /mob/living/silicon/ai/ShiftClickOn(var/atom/A)
-	A.AIShiftClick(src)
+	if(isturf(A)) // Skyrat edit
+		var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A 
+		if(airlock)
+			airlock.AIShiftClick(src)
+		A.AIShiftClick(src)
+	A.AIShiftClick(src) // End of skyrat edit
```
Comment (Azarak) [inline code/_onclick/ai.dm:125]:
```suggestion
	if(isturf(A)) // Skyrat edit
		var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A 
		if(airlock)
			airlock.AIShiftClick(src)
	else
		A.AIShiftClick(src) // End of skyrat edit
```

File: code/_onclick/ai.dm [hunk 7]
```diff
@@ -109,11 +109,29 @@
...
+
 /mob/living/silicon/ai/CtrlClickOn(var/atom/A)
-	A.AICtrlClick(src)
+	if(isturf(A)) // Skyrat edit
+		var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A 
+		if(airlock)
+			airlock.AICtrlClick(src)
+		A.AICtrlClick(src)
+	A.AICtrlClick(src) // End of skyrat edit
```
Comment (Azarak) [inline code/_onclick/ai.dm:133]:
```suggestion
	if(isturf(A)) // Skyrat edit
		var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A 
		if(airlock)
			airlock.AICtrlClick(src)
	else
		A.AICtrlClick(src) // End of skyrat edit
```

File: code/_onclick/cyborg.dm [hunk 8]
```diff
@@ -104,11 +104,29 @@
 //Give cyborgs hotkey clicks without breaking existing uses of hotkey clicks
 // for non-doors/apcs
 /mob/living/silicon/robot/CtrlShiftClickOn(atom/A)
-	A.BorgCtrlShiftClick(src)
+	if(isturf(A)) // Skyrat edit
+		var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A // Skyrat edit
+		if(airlock)
+			airlock.BorgCtrlShiftClick(src)
+		A.BorgCtrlShiftClick(src)
+	A.BorgCtrlShiftClick(src) // End of skyrat edit
```
Comment (Azarak) [inline code/_onclick/cyborg.dm:112]:
```suggestion
	if(isturf(A)) // Skyrat edit
		var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A // Skyrat edit
		if(airlock)
			airlock.BorgCtrlShiftClick(src)
	else
		A.BorgCtrlShiftClick(src) // End of skyrat edit
```

File: code/_onclick/cyborg.dm [hunk 9]
```diff
@@ -104,11 +104,29 @@
...
+
 /mob/living/silicon/robot/ShiftClickOn(atom/A)
-	A.BorgShiftClick(src)
+	if(isturf(A)) // Skyrat edit
+		var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A // Skyrat edit
+		if(airlock)
+			airlock.BorgShiftClick(src)
+		A.BorgShiftClick(src)
+	A.BorgShiftClick(src) // End of skyrat edit
```
Comment (Azarak) [inline code/_onclick/cyborg.dm:120]:
```suggestion
	if(isturf(A)) // Skyrat edit
		var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A // Skyrat edit
		if(airlock)
			airlock.BorgShiftClick(src)
	else
		A.BorgShiftClick(src) // End of skyrat edit
```

File: code/_onclick/cyborg.dm [hunk 10]
```diff
@@ -104,11 +104,29 @@
...
+	
 /mob/living/silicon/robot/CtrlClickOn(atom/A)
-	A.BorgCtrlClick(src)
+	if(isturf(A)) // Skyrat edit
+		var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A // Skyrat edit
+		if(airlock)
+			airlock.BorgCtrlClick(src)
+		A.BorgCtrlClick(src)
+	A.BorgCtrlClick(src) // End of skyrat edit
```
Comment (Azarak) [inline code/_onclick/cyborg.dm:128]:
```suggestion
	if(isturf(A)) // Skyrat edit
		var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in A // Skyrat edit
		if(airlock)
			airlock.BorgCtrlClick(src)
	else
		A.BorgCtrlClick(src) // End of skyrat edit
```

**Review by Azarak — CHANGES_REQUESTED** [review]

**Review by Azarak — APPROVED** [review]

**Pull request merged**